pkg/client/userd/grpc.go

```diff
@@ -498,22 +502,91 @@ func (s *Service) ListCommands(ctx context.Context, _ *empty.Empty) (groups *rpc
 	return
 }
 
-func stdinPump(ctx context.Context, cmdStream rpc.Connector_RunCommandServer, wr io.Writer) {
-	for ctx.Err() == nil {
-		cr, err := cmdStream.Recv()
-		if err != nil {
-			if err != io.EOF && ctx.Err() == nil {
-				dlog.Errorf(ctx, "command failed to read stdin: %v", err)
+func needsPTY(cmd *cobra.Command) bool {
+	if runtime.GOOS == "windows" || cmd.Name() != "intercept" {
+		// intercept is the only known command that might need a PTY, and never on Windows
+		return false
+	}
+	flags := cmd.Flags()
+	args := flags.Args()
+	if len(args) <= 1 {
+		// No extra args, so no external command will be executed
+		return false
+	}
+	args = args[1:]
+	if dr := flags.Lookup("docker-run"); dr == nil || dr.Value.String() == "false" {
+		// Don't know what type of command this is, so assume that a PTY is needed
+		return true
+	}
+
+	// This is a docker run. It needs a PTY if used with -it, but only then.
+	hasI := false
+	hasT := false
+	for _, arg := range args {
+		if len(arg) >= 2 && arg[0] == '-' && arg[1] != '-' {
+			arg = arg[1:]
+			if !hasI {
+				hasI = strings.ContainsRune(arg, 'i')
+			}
+			if !hasT {
+				hasT = strings.ContainsRune(arg, 't')
+			}
+			if hasI && hasT {
```

Comment:
So I think we only need `if hasT {}` here since `-t` / `--tty` alone is used to request a pty for the container, `-i`/`--interactive` is specific to stdin but we should still support streaming commands.

Reply:
We actually don't need a PTY unless `-i` is used. Using only `-t` works fine without a PTY. The long forms `--tty` and `--interactive` must be supported though. I'll fix that.

Reply:
There's a good text regarding `-i` and `-t` here: https://stackoverflow.com/a/54254380/328043